### ISSUE TITLE
Custom TCTI

### ISSUE THREAD

**Issue by stefanodevenuto** [parallaxsecond/rust-tss-esapi#533]
Hi everyone!
I was wondering if in the library there's a way to indicate a custom TCTI to be used, like with the `-T` option using tpm2_tools, and, if not, if it would require major changes (the tools leverage dlopen() to load the .so provided)

Thanks a lot!

**Comment by Superhepper** [comment]
No, the TCTI options that currently exist are explicitly defined so there is no option that would allow for that currently I think.

**Comment by Superhepper** [comment]
I am not sure how the tpm2-tools do that but it would probably not be a major issue to implement the functionality but rather a bit of a hassle to add some kind of CI pipeline to test it.

**Comment by wiktor-k** [comment]
Hm... can't [`TctiNameConf::from_str`](https://docs.rs/tss-esapi/latest/tss_esapi/tcti_ldr/enum.TctiNameConf.html#impl-FromStr-for-TctiNameConf) be used for that?

I see it's used as part of [`TctiNameConf::from_environment_variable`](https://github.com/parallaxsecond/rust-tss-esapi/pull/545/files#diff-a2b5450c5b3a1346f3e0101a20b49af03c04367b1b384bfda627e424ea33e627R174)... or maybe I'm mixing it up with something else? :thinking:

**Comment by KarstenB** [comment]
Yes, please add a catch-all variant. I am currently unable to connect to a swtpm on a unix socket, which is really unfortunate.

**Comment by Superhepper** [comment]
What version are you using? Because If I am not mistaken on main and therefor in the upcoming 8.0.0 release. It is possible to use unix sockets.

**Comment by KarstenB** [comment]
I am on 7.7.0 which definitely doesn't support that just yet, but 8.0.0.alpha.2 does. I am switching to that for now, thanks for the tip.
